### PR TITLE
deciso/OPNWAF: Include new features into OPNWAF documentation

### PR DESCRIPTION
**Pull request by Monviech**
This improves the docs with a mix of new features that come for OPNWAF.

Since these features are spread across 2 versions of OPNsense Business edition, these features have been marked with their first appearance in present or future versions, so its clear for the user if they have them or not depending on their OPNsense version.

### CODE REVIEW

File: source/vendor/deciso/opnwaf.rst [hunk 1]
```diff
@@ -67,53 +76,75 @@ After installation, the module itself should be enabled by default.
...
 
-
 The section above defines the port the virtual server will listen on. Remember, in order to use ACME (Let's encrypt) this should either
 be 443 or the traffic should be forwarded from port 443 to the port defined here.
 
 .. Note::
 
     Port numbers don't have to be unique when more virtual servers are defined as the hostname correctly identifies the
-    location.
+    location. Yet, there can't be duplicate hostnames, these have to be unique.
```
Comment (fichtner) [inline source/vendor/deciso/opnwaf.rst:147]: 1. do not use contractions in technical documentation: can't -> cannot, etc.
2. try to make it snappy: Hostnames, however, have to be unique.

File: source/vendor/deciso/opnwaf.rst [hunk 2]
```diff
@@ -124,24 +155,51 @@ be 443 or the traffic should be forwarded from port 443 to the port defined here
...
+~~~~~~~~~~~~~~~~~~~~~~~~
+
 The virtual server itself doesn't provide much content to the user other than offering a page telling access is prohibited,
 so the next step is to map directories to external locations. These can be defined in the "Locations" Grid underneath
 the Virtual servers.
 
+There are two different modes for locations:
+
+#. | ProxyPass, which Reverse Proxies the HTTP traffic
+#. | Redirect, which creates a HTTP redirect (available as of 24.10)
```
Comment (fichtner) [inline source/vendor/deciso/opnwaf.rst:169]: Avoid version numbers. This information goes stale so fast and adds no value.

File: source/vendor/deciso/opnwaf.rst [hunk 3]
```diff
@@ -173,14 +262,57 @@ This should show a page similar to the one below:
     :width: 50%
 
 
+When deploying web protection for virtual servers, it is a good idea to start with the `Detection Only` setting that can be set per virtual server.
```
Comment (fichtner) [inline source/vendor/deciso/opnwaf.rst:265]: avoid fillers like "it is a good idea". The sentence works fine without it and it's faster reading :)

```suggestion
When deploying web protection for virtual servers, start with the `Detection Only` setting that can be set per virtual server.
```

File: source/vendor/deciso/opnwaf.rst [hunk 4]
```diff
@@ -173,14 +262,57 @@ This should show a page similar to the one below:
...
+and would block connections.
+
+If they are false positives, the rule IDs can be set as excemptions with the option `Disable Security Rules by ID`. Search the rules
+in the dropdown, and select multiple ones you want to exclude.
+
+After this configuration, set the Web Protection to `On (default)` to enable it. The web application should now be configured for production.
+If there are still errors, repeat the above steps.
+
+.. Attention::
+    Do not exclude too many rules. These matches could be a potential misconfiguration of the web application behind the WAF. Only exclude rules
```
Comment (fichtner) [inline source/vendor/deciso/opnwaf.rst:278]: Ah, contraction free sentence :)

**Comment by Monviech** [comment]
Lets still wait with this one I have some more things to add.

**Comment by Monviech** [comment]
This one should be ok.

File: source/vendor/deciso/opnwaf.rst [hunk 5]
```diff
@@ -158,8 +218,145 @@ destinations this path should map to (for example you could point to a public se
...
+EcpVirtualDirectory              ``mail.example.com/ecp``
+WebServicesVirtualDirectory      ``mail.example.com/EWS/Exchange.asmx``
+ActiveSyncVirtualDirectory       ``mail.example.com/Microsoft-Server-ActiveSync``
+OabVirtualDirectory              ``mail.example.com/OAB``
+MapiVirtualDirectory             ``mail.example.com/mapi``
+OutlookAnywhere                  ``mail.example.com/rpc`` - `ExternalClientAuthenticationMethod` set to `Negotiate`
+ClientAccessService              ``autodiscover.example.com/Autodiscover/Autodiscover.xml``
+================================ ========================================================================================
+
+When using a self-signed certificate, it must be imported into :menuselection:`System->Trust->Authorities`.
```
Comment (AdSchellevis) [inline source/vendor/deciso/opnwaf.rst:297]:
```suggestion
When using a self-signed certificate, the authority for the certificate must be imported into :menuselection:`System->Trust->Authorities`.
```

**Pull request merged**